File: backend/schemas/airtable/project_listing_table.py
# -*- coding: utf-8 -*-
# -*- Python Version: 3.11 -*-

"""Pydantic Schema: AirTable | Master Project List"""


from pydantic import BaseModel


class AT_ProjectListingRecordFieldsSchema(BaseModel):
    PROJECT_NUMBER: str
    PROJECT_NAME: str
    APP_ID: str
    TABLE_ID: str


class AT_ProjectListingRecordSchema(BaseModel):
    id: str
    createdTime: str
    fields: AT_ProjectListingRecordFieldsSchema


class AT_ProjectListingTableSchema(BaseModel):
    """The listing of the Active Projects from AirTable."""

    records: list[AT_ProjectListingRecordSchema]

File: src/types/airtable/project_listing_table.ts

type ProjectListingRecordFields = {
    PROJECT_NUMBER: string;
    PROJECT_NAME: string;
    APP_ID: string; // The Airtable App-ID for the specified project
    TABLE_ID: string; // The Airtable Table-ID for the specified project
}

type ProjectListingRecord = {
    id: string;
    createdTime: string;
    fields: ProjectListingRecordFields;
}

export type ProjectListingTable = {
    records: ProjectListingRecord[];
}